mongo/mongo_test.go

```diff
@@ -72,52 +72,40 @@ func TestMongoHelpers(t *testing.T) {
 	t.Run("transform and ensure ID", func(t *testing.T) {
 		t.Run("newly added _id should be first element", func(t *testing.T) {
 			doc := bson.D{{"foo", "bar"}, {"baz", "qux"}, {"hello", "world"}}
-			want := bsonx.Doc{
-				{"_id", bsonx.Null()}, {"foo", bsonx.String("bar")},
-				{"baz", bsonx.String("qux")}, {"hello", bsonx.String("world")},
-			}
 			got, id, err := transformAndEnsureID(bson.DefaultRegistry, doc)
 			assert.Nil(t, err, "transformAndEnsureID error: %v", err)
 			oid, ok := id.(primitive.ObjectID)
 			assert.True(t, ok, "expected returned id type %T, got %T", primitive.ObjectID{}, id)
-			want[0] = bsonx.Elem{"_id", bsonx.ObjectID(oid)}
-			assert.Equal(t, got, want, "expected document %v, got %v", got, want)
-		})
-		t.Run("existing _id should be first element", func(t *testing.T) {
-			doc := bson.D{{"foo", "bar"}, {"baz", "qux"}, {"_id", 3.14159}, {"hello", "world"}}
```

Comment:
Can we add this test back, to test that an existing `_id` will remain in the same order? That way, if we do change the ordering behavior later, we will be informed by tests.